src/astro_databricks/operators/workflow.py

```diff
@@ -345,42 +365,39 @@ def __init__(
 
     def __exit__(self, _type, _value, _tb):
         """Exit the context manager and add tasks to a single _CreateDatabricksWorkflowOperator."""
-        roots = self.roots
+        roots = list(self.get_roots())
+        tasks = flatten_node(self)
 
         # For Airflow versions <2.3, the `dag` attribute is unassociated, and hence we need to add that.
         if not hasattr(self, "dag"):
             from airflow.models.dag import DagContext
 
             self.dag = DagContext.get_current_dag()
 
-        create_databricks_workflow_task: _CreateDatabricksWorkflowOperator = (
-            _CreateDatabricksWorkflowOperator(
-                dag=self.dag,
-                task_id="launch",
-                databricks_conn_id=self.databricks_conn_id,
-                job_clusters=self.job_clusters,
-                existing_clusters=self.existing_clusters,
-                extra_job_params=self.extra_job_params,
-            )
+        create_databricks_workflow_task = _CreateDatabricksWorkflowOperator(
+            dag=self.dag,
+            task_group=self,
+            task_id="launch",
+            databricks_conn_id=self.databricks_conn_id,
+            job_clusters=self.job_clusters,
+            existing_clusters=self.existing_clusters,
+            extra_job_params=self.extra_job_params,
         )
 
-        for task in roots:
+        for task in tasks:
             if not (
                 hasattr(task, "convert_to_databricks_workflow_task")
                 and callable(task.convert_to_databricks_workflow_task)
             ):
                 raise AirflowException(
                     f"Task {task.task_id} does not support conversion to databricks workflow task."
                 )
-            create_databricks_workflow_task.set_upstream(
-                task_or_task_list=list(task.upstream_list)
-            )
 
-        for task_id, task in self.children.items():
-            if task_id != f"{self.group_id}.launch":
-                create_databricks_workflow_task.relevant_upstreams.append(task_id)
-                create_databricks_workflow_task.add_task(task)
-                task.databricks_metadata = create_databricks_workflow_task.output
+            create_databricks_workflow_task.relevant_upstreams.append(task.task_id)
```

Comment:
Do we no longer need to worry about the launch task?

Reply:
Not anymore! We get the list of tasks before creating the launch task, so this list should never have the launch task.